### ISSUE TITLE
non-nested EDA unit test failing

### ISSUE THREAD

**Issue by smjenness** [statnet/EpiModel#545]
This unit test now failing on Actions with the current Statnet4 stack.

```r
test_that("environment handling in non-nested EDA", {
  nw <- network_initialize(n = 1000)
  nw %v% "race" <- rep(letters[1:5], length.out = 1000)
  nw %v% "age" <- rep(1:3, length.out = 1000)

  make_formula <- function() {
    a <- "age"
    ff <- ~offset(edges) + offset(nodematch(a, diff = TRUE))
    ff
  }

  ff <- make_formula()

  coef_diss <- dissolution_coefs(ff, c(55, 45, 65, 80))

  r <- "race"

  netest_1 <- netest(nw = nw,
                     formation = ~edges + nodematch(r, diff = TRUE),
                     target.stats = c(500, 50, 50, 90, 40, 60),
                     coef.diss = coef_diss,
                     nested.edapprox = FALSE)

  expect_error(netdx_1 <- netdx(netest_1, nsims = 2, nsteps = 5, dynamic = TRUE),
               "object 'a' not found")

  netest_2 <- netest(nw = nw,
                     formation = ~edges + nodematch(r, diff = TRUE),
                     target.stats = c(500, 50, 50, 90, 40, 60),
                     coef.diss = coef_diss,
                     nested.edapprox = FALSE,
                     from.new = "a")

  expect_error(netdx_2 <- netdx(netest_2, nsims = 2, nsteps = 5, dynamic = TRUE), NA)

  make_formula_2 <- function() {
    x <- "race"
    ff <- ~offset(edges) + offset(nodematch(x, diff = TRUE))
    ff
  }

  ff_2 <- make_formula_2()

  coef_diss_2 <- dissolution_coefs(ff_2, c(30, 40, 50, 25, 35, 55))

  expect_error(netest_3 <- update_dissolution(netest_2, coef_diss_2,
                                              nested.edapprox = TRUE),
               "Term options for one or more terms in dissolution model")
  netest_3 <- update_dissolution(netest_2, coef_diss_2, nested.edapprox = FALSE)
  expect_error(netdx_3 <- netdx(netest_3, nsims = 2, nsteps = 5, dynamic = TRUE),
               "object 'x' not found")
  netest_4 <- update_dissolution(netest_2, coef_diss_2, nested.edapprox = FALSE,
                                 from.new = "x")
  expect_error(netdx_4 <- netdx(netest_4, nsims = 2, nsteps = 5, dynamic = TRUE), NA)
})
```

```r
 ── Warning (Line 34): environment handling in non-nested EDA ─────────────────────
longer object length is not a multiple of shorter object length
Backtrace:
 1. testthat::expect_error(...)
 6. EpiModel::netdx(netest_2, nsims = 2, nsteps = 5, dynamic = TRUE)

── Warning (Line 54): environment handling in non-nested EDA ─────────────────────
longer object length is not a multiple of shorter object length
Backtrace:
 1. testthat::expect_error(...)
 6. EpiModel::netdx(netest_4, nsims = 2, nsteps = 5, dynamic = TRUE)
```

**Comment by smjenness** [comment]
I fixed the other two failing unit tests in 80f26b66917a4225946a6ffeac97337d57758dfc

**Comment by chad-klumb** [comment]
What I get are warnings:

```
 Warning message:
In x$coef.diss$duration^2 * dgeom(2:(nsteps + 1), 1/x$coef.diss$duration) :
  longer object length is not a multiple of shorter object length
```

and

```
 Warning message:
In duration.obs + pages_imptd :
  longer object length is not a multiple of shorter object length
```

which appear to be `netdx` related, e.g.

https://github.com/statnet/EpiModel/issues/529#issue-896974171

**Comment by smjenness** [comment]
ok, I can fix these... reassigning to me.

**Comment by smjenness** [comment]
Closed with c3cb8595392c0683023f0baae8d66a06b2ce04e4
